### PR TITLE
Guarantee local-first CloudSync integrity

### PR DESCRIPTION
**Pull request by ComputelessComputer**
Drain and defer CloudSync around recording, chat, E2EE, auth teardown, transcript persistence, and summary writes. Bound native cancellation and initialization, add regression coverage, and document exact-SHA Dev, staging, and installed-stable QA with CLI podcast AEC testing.

Verification: 2,156 desktop tests, 126 web tests, full workspace TypeScript and Rust checks, db-app/db-core/plugin suites, clippy, formatting, and QA helper tests.

---
This is **part 4 of 5 in a stack** made with GitButler:
- <kbd>&nbsp;5&nbsp;</kbd> #6241
- <kbd>&nbsp;4&nbsp;</kbd> #6243 👈 
- <kbd>&nbsp;3&nbsp;</kbd> #6237
- <kbd>&nbsp;2&nbsp;</kbd> #6233
- <kbd>&nbsp;1&nbsp;</kbd> #6226

---

> [!NOTE]
> **High Risk**
> Changes core chat persistence, CloudSync lease timing, auth callback handling, and release evidence gates; regressions could cause sync races, lost chat history on regen, or blocked releases.
> 
> **Overview**
> **Chat and CloudSync** now hold native activity leases through preflight persistence, tracked side writes, and unmount cleanup (including `flushDatabaseWritesByPrefix`). Regeneration **atomically replaces** assistant rows via `replaceChatMessage` instead of tombstoning before transport; failed/empty/aborted retries keep the prior durable answer. The CloudSync activity controller adds **`runWithLease`**, retained preflight repair, and stricter dispose/release semantics so sync cannot overlap SQLite writes.
> 
> **Auth** adds **`createAuthCallbackHandler`** to dedupe in-flight and recent duplicate OAuth deeplink callbacks while allowing retries after failure.
> 
> **Release QA** expands the critical-UX skill (sign-in/out, chat overlap deferral, exact-SHA Dev → staging → installed stable), pins candidates with **`ANARLOG_QA_GIT_SHA`** / GitButler resolver, fingerprints **`git_head_sha`/`git_dirty`** in the native Dev manifest, and requires built-in Mac audio before launch. **`.claude/skills`** become thin pointers to **`.agents/skills`**.
> 
> **Tests** add broad coverage for chat session provider, CloudSync activity edge cases, auth deeplink, and query atomic replace.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ebc7428b8a685e5aee6d62ee7479e728c3b8e330. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by ComputelessComputer** [comment]
cursor review

File: apps/desktop/src/chat/components/session-provider.tsx [hunk 1]
```diff
@@ -329,19 +351,39 @@ function ChatSessionLifecycle({
...
+        ]);
+        try {
+          await flushDatabaseWritesByPrefix(["chat:"]);
+        } catch (error) {
+          console.error("Failed to flush chat writes during cleanup", error);
+        }
+      });
+      void chatCloudsyncActivity.dispose(cleanup);
+      pendingTransportPreflightsRef.current.clear();
+      pendingRegenerationTombstonesRef.current.clear();
```
[the remark this answers is not in the document]
Comment (ComputelessComputer) [reply apps/desktop/src/chat/components/session-provider.tsx:427]: Fixed in 562fedc. Durable send persistence is now bound to the acquired CloudSync lease and survives disposal, while canceled regeneration preflights remain disposable and transport never starts after cleanup. Added deferred-acquisition regressions at the guard and provider layers; focused 45/45, full desktop 2,160/2,160, desktop/workspace typechecks pass.

**Comment by ComputelessComputer** [comment]
cursor review

**Comment by ComputelessComputer** [comment]
cursor review

**Comment by ComputelessComputer** [comment]
cursor review

File: apps/desktop/src/chat/store/cloudsync-activity.ts [hunk 2]
```diff
@@ -229,31 +287,46 @@ export function createChatCloudsyncActivityController({
...
+      preflightError = error;
+    } finally {
+      lease.preflightPending = false;
+      finishLeaseIfSettled(lease);
+    }
+
+    if (preflightFailed) {
+      consumeCallback(lease);
+      throw preflightError;
+    }
```
[the remark this answers is not in the document]
Comment (ComputelessComputer) [reply apps/desktop/src/chat/store/cloudsync-activity.ts:460]: This is intentional lease coalescing, not a leaked pause. A failed preflight marks its attempt finished and schedules the normal 750 ms trailing release. A subsequent start cancels that timer because its new live/acquiring lease must keep CloudSync paused; when that attempt settles, the idle release releases both native attempt keys. If the second attempt remains pending, that second lease independently requires the pause. The regression `keeps guard failures bound to their own completion callback` covers fail → retry → finish and asserts both attempt keys are released. No change needed.

**Comment by ComputelessComputer** [comment]
cursor review

File: apps/desktop/src/chat/store/cloudsync-activity.ts [hunk 3]
```diff
@@ -228,84 +426,198 @@ export function createChatCloudsyncActivityController({
...
-    finishLease(lease);
-    removeFromLogicalQueue(lease);
-  };
-
   const finish = (logicalKey: string) => {
-    const queue = logicalQueues.get(logicalKey);
-    const lease = queue?.find((candidate) => !candidate.callbackConsumed);
+    const lease =
+      claimActiveFinishLease(logicalKey) ??
+      claimConsumedFinishLease(logicalKey);
```
[the remark this answers is not in the document]
Comment (ComputelessComputer) [reply apps/desktop/src/chat/store/cloudsync-activity.ts]: Fixed in ebc7428: `finish` now consumes only an active callback owner, so retained durable preflights remain queued for the next leased repair. Added a regression covering acquisition failure, early finish, and successful replay.

**Comment by ComputelessComputer** [comment]
cursor review
